Install zip.js library via npm

lachnerm opened the pull request:
This PR adds the zip.js library to the project via npm and removes the custom installation via the vendors folder. If I've missed any references to the custom installation in any other scripts, I will remove them too of course.
Generally the library didn't change much. They adapted some newer JS techniques like Promises, but the functionality is pretty much the same. I therefore mainly adjusted our code to the new API in the `LinkOverlay` component.

Fixes #715.

PhilippWendler reviewed (commented):
I think you got all cases where we can simplify config, thanks!

According to the output of `npm run build`, the gzipped bundle size would now be 17 kB larger. I suspect this is because zip.js includes several features (zip-fs, deflate, and the web-worker stuff) that we don't need. Could you have a look at whether there is a way to change how we include zip.js such that only the relevant parts get included (e.g., by not including the full `zip.min.js` but only individual files or so)? If there is no way to do this nicely right now but we could do it if something is changed in the library, we can also ask its developer, it is quite active and responsive right now.

lachnerm commented on benchexec/tablegenerator/react-table/src/tests/setup/setup.js:
```diff
@@ -8,6 +8,7 @@
 // enzyme
 import { configure } from "enzyme";
 import Adapter from "enzyme-adapter-react-16";
+window.crypto = require("@trust/webcrypto");
```
Yes sorry, I missed this. I've added a comment.

lachnerm commented on benchexec/tablegenerator/react-table/src/components/LinkOverlay.js:
```diff
@@ -10,10 +10,13 @@ import ReactModal from "react-modal";
 import classNames from "classnames";
 import path from "path";
 import TaskDefinitionViewer from "./TaskDefinitionViewer.js";
+const zip = require("../../node_modules/@zip.js/zip.js/dist/zip.min.js");
```
I didn't get it to work with the version I used at that time, so i stumbled across this "solution" in the issues of the library. With the newer version, it works now any way though, so I can use a regular `import` statement.

lachnerm commented on benchexec/tablegenerator/react-table/src/components/LinkOverlay.js:
```diff
@@ -96,7 +99,7 @@ export default class LinkOverlay extends React.Component {
           this.setState({ content });
         } else {
-          this.loadFileFromZip(url);
+          throw Error(`Received response status ${response.status}`);
```
There is no actual change in functionality here. The statement is within a `try` block, that calls the `loadFileFromZip` function when an error occurs - and when entering this else block, the response has returned an error (or at least that something isn't working). So I thought it would be more intuitive to also throw an error in this case and call the `loadFileFromZip` function from the `except` block. Also this helped me a bit during debugging, because there was only a single entry for this function with different errors, showing why exactly the previous attempt failed.

PhilippWendler replied:
Nice!

lachnerm commented:
> According to the output of `npm run build`, the gzipped bundle size would now be 17 kB larger. I suspect this is because zip.js includes several features (zip-fs, deflate, and the web-worker stuff) that we don't need. Could you have a look at whether there is a way to change how we include zip.js such that only the relevant parts get included (e.g., by not including the full `zip.min.js` but only individual files or so)? If there is no way to do this nicely right now but we could do it if something is changed in the library, we can also ask its developer, it is quite active and responsive right now.

Currently we're importing the `@zip.js/zip.js/dist/zip.min.js` file, which seems to be the smallest import that's possible. This way we can work around some imports and decrease the size of the bundled files a little bit compared to the normal import. The other files either don't fully cover our needs or don't decrease the bundle sizes. And it seems that is's not easily possible to remove more imports - I've browsed through the libary and most of the other parts seem to be bound tightly to the core features.

PhilippWendler commented:
I see. Would it make sense to ask the library developer for something that would help us reduce the size? Would we have a concrete suggestion of what they could add or how they could refactor the code?

lachnerm commented:
> I see. Would it make sense to ask the library developer for something that would help us reduce the size? Would we have a concrete suggestion of what they could add or how they could refactor the code?

Hm, I actually don't think there's any reasonable change that could be done. We're already working around importing the `inflate` and `deflate` stuff. The file we import imports all the stuff from the `zip-core.js` file as well as the `configure` function to deactivate the web workers. From this file on, all the components of the library itself (ZipReader, ZipWriter, BlobReader, HttpReader,...) as well as smaller things such as error messages are gathered. I didn't find a large file that could be easily excluded, so basically excluding anything from here would only be cherry-picking what we in particular need for our use case - so I don't think there's anything reasonable that we could ask the developer.

gildas-lormeau commented:
@lachnerm I tried to improve things on `master` by adding `zip-no-worker*` files here https://github.com/gildas-lormeau/zip.js/tree/master/lib and https://github.com/gildas-lormeau/zip.js/tree/master/dist. I hope this can help you.

PhilippWendler commented:
@gildas-lormeau Wow, thanks for your incredible support! Indeed, `zip-no-worker-inflate.min.js` seems like it is exactly what we need. I tried it out quickly now and it worked and saved us about 40 kB.

@lachnerm Could you please adopt this change (and bump the version to 2.2.9) and then test it properly and finish the PR? Thanks.

gildas-lormeau commented:
@PhilippWendler You're welcome. I had exactly the same need in [SingleFileZ](https://github.com/gildas-lormeau/SingleFileZ) actually ;)

lachnerm commented:
@gildas-lormeau Thank you very much for the support! This is indeed exactly what we needed and reduced the bundle size by a great deal, even compared to the file from the `dist` folder we imported before.

@PhilippWendler I've updated the library to the newest version, changed the import to the newly added `zip-no-worker-inflate.min.js` file and confirmed that everything is still working the same way it did before.
